pkg/virt-handler/vm_test.go

```diff
@@ -1465,66 +1469,136 @@ var _ = Describe("VirtualMachineInstance", func() {
+			vmi.UID = vmiTestUUID
+			vmi.ObjectMeta.ResourceVersion = "1"
+			interface_name := "interface_name"
 			vmi.Spec.Networks = []v1.Network{
-				v1.Network{
-					Name: networkName,
-					NetworkSource: v1.NetworkSource{
-						Pod: &v1.PodNetwork{
-							VMNetworkCIDR: "1.1.1.1",
-						},
-					},
+				{
+					Name:          interface_name,
+					NetworkSource: v1.NetworkSource{Pod: &v1.PodNetwork{}},
+				},
+			}
+			vmi.Status.Phase = v1.Running
+			vmi.Status.Interfaces = make([]v1.VirtualMachineInstanceNetworkInterface, 0)
+
+			podCacheInterface := network.PodCacheInterface{
+				Iface: &v1.Interface{
+					Name: interface_name,
 				},
+				PodIP: "1.1.1.1",
 			}
+			podJson, err := json.Marshal(podCacheInterface)
+			Expect(err).ToNot(HaveOccurred())
+			err = os.MkdirAll(fmt.Sprintf(util.VMIInterfaceDir, vmi.UID), 0755)
+			Expect(err).ToNot(HaveOccurred())
+			vmiInterfacepath := fmt.Sprintf(util.VMIInterfacepath, vmi.UID, interface_name)
+			f, err := os.Create(vmiInterfacepath)
+			Expect(err).ToNot(HaveOccurred())
+			_, err = f.WriteString(string(podJson))
+			Expect(err).ToNot(HaveOccurred())
+			f.Close()
+
+			mockWatchdog.CreateFile(vmi)
+			domain := api.NewMinimalDomainWithUUID("testvmi", vmiTestUUID)
+			domain.Status.Status = api.Running
+			vmiFeeder.Add(vmi)
+			domainFeeder.Add(domain)
+
+			client.EXPECT().SyncVirtualMachine(vmi, gomock.Any()).Do(func(vmi *v1.VirtualMachineInstance, options *cmdv1.VirtualMachineOptions) {
+				Expect(options.VirtualMachineSMBios.Family).To(Equal(virtconfig.SmbiosConfigDefaultFamily))
+				Expect(options.VirtualMachineSMBios.Product).To(Equal(virtconfig.SmbiosConfigDefaultProduct))
+				Expect(options.VirtualMachineSMBios.Manufacturer).To(Equal(virtconfig.SmbiosConfigDefaultManufacturer))
+			})
+
 			vmiInterface.EXPECT().Update(gomock.Any()).Do(func(arg interface{}) {
 				Expect(len(arg.(*v1.VirtualMachineInstance).Status.Interfaces)).To(Equal(1))
-				Expect(arg.(*v1.VirtualMachineInstance).Status.Interfaces[0].Name).To(Equal(networkName))
-				Expect(arg.(*v1.VirtualMachineInstance).Status.Interfaces[0].IP).To(Equal(podIp))
-				Expect(arg.(*v1.VirtualMachineInstance).Status.Interfaces[0].IPs[0]).To(Equal(podIp))
+				Expect(arg.(*v1.VirtualMachineInstance).Status.Interfaces[0].Name).To(Equal(podCacheInterface.Iface.Name))
+				Expect(arg.(*v1.VirtualMachineInstance).Status.Interfaces[0].IP).To(Equal(podCacheInterface.PodIP))
+				Expect(arg.(*v1.VirtualMachineInstance).Status.Interfaces[0].IPs[0]).To(Equal(podCacheInterface.PodIP))
+
 			}).Return(vmi, nil)
+
 			controller.Execute()
 		})
 
-		It("should only report the pod network in status", func() {
-			vmi := NewPendingVirtualMachine("testvmi")
-			vmi.Status.Phase = v1.Scheduling
-			pod := NewPodForVirtualMachine(vmi, k8sv1.PodRunning)
-
-			addVirtualMachine(vmi)
-			podFeeder.Add(pod)
-
-			networkName := "test net"
+		It("Should update existing interface with latest IP", func() {
```

Comment:
I would change it to - 
It("Should update masquerade interface with the pod IP" ...

Reply:
changed